warehouse/api/echo.py

```diff
@@ -0,0 +1,126 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import typing
+
+from pyramid.httpexceptions import HTTPAccepted, HTTPBadRequest
+from pyramid.view import view_config
+
+from warehouse.authnz import Permissions
+from warehouse.observations.models import OBSERVATION_KIND_MAP, ObservationKind
+
+if typing.TYPE_CHECKING:
+    from pyramid.request import Request
+
+    from warehouse.packaging.models import Project
+
+
+# TODO: Move this to a more general-purpose API view helper module
+def api_v0_view_config(**kwargs):
+    """
+    A helper decorator that is used to create a view configuration that is
+    useful for API version 0 views. Usage:
+
+        @api_v0_view_config(
+            route_name="api.projects",
+            permission=Permissions.API...,
+        )
+        def ...
+    """
+
+    # Prevent developers forgetting to set a permission
+    if "permission" not in kwargs:
+        raise TypeError("`permission` keyword is is required")
+
+    # Set defaults for API views
+    kwargs.update(
+        accept="application/vnd.pypi.api-v0-beta+json",
+        renderer="json",
+        require_csrf=False,
+        # TODO: Can we apply a macaroon-based rate limiter here,
+        #  and how might we set specific rate limits for user/project owners?
+    )
+
+    def _wrapper(wrapped):
+        return view_config(**kwargs)(wrapped)
+
+    return _wrapper
+
+
+@api_v0_view_config(
+    route_name="api.echo",
+    permission=Permissions.APIEcho,
+)
+def api_echo(request: Request):
+    return {
+        "accept": "I accept version 1 of the API",
```

Comment:
thought: use our view_config to bind an `api_version` string to the `Request` that would allow this to be

```
        "accept": f"I accept version {request.api_version} of the API",
```